modules/internal/ChapelRange.chpl

```diff
@@ -2176,10 +2176,57 @@ operator :(r: range(?), type t: range(?)) {
   // An unbounded range iterator (for all strides)
   pragma "no doc"
   pragma "order independent yielding loops"
-  iter range.these() where boundedType != BoundedRangeType.bounded {
+  iter range.these() where boundedType == BoundedRangeType.boundedNone {
+    compilerError("iteration over a range with no bounds");
+  }
 
-    if boundedType == BoundedRangeType.boundedNone then
-      compilerError("iteration over a range with no bounds");
+  // An unbounded range iterator (for all strides)
+  pragma "no doc"
+  pragma "order independent yielding loops"
+  iter range.these() where boundedType == BoundedRangeType.boundedLow {
+
+    if boundsChecking {
+      if ! this.hasFirst() then
+        HaltWrappers.boundsCheckHalt("iteration over range that has no first index");
+
+      if this.isAmbiguous() then
+        HaltWrappers.boundsCheckHalt("these -- Attempt to iterate over a range with ambiguous alignment.");
+    }
+    // This iterator could be split into different cases depending on the
+    // stride like the bounded iterators. However, all that gets you is the
+    // ability to use low/alignedLow over first. The additional code isn't
+    // worth it just for that.
+    var i: intIdxType;
+    const start = chpl__idxToInt(this.first);
+    const end = if isBoolType(idxType)
```

Comment:
Are you thinking that `max(bool)` might give us 1 and that `max(myEnum)` might give us `size-1`?  It looks like `max(bool)` gives us `true` and that `max(myEnum)` is not currently supported (but if it were, I'd still expect it to give `myEnum.last` rather than `myEnum.size - 1`).  This relates to the difference between `idxType` and `intIdxType` in the current range implementation, along with the `chpl__idxToInt()` helpers and the like.  These are here so that we can run loops over raw integers directly, and then convert to things like bools or enums at yield points.